File: go/internal/replay/parserfixture/testdata/trees/kotlin/OrderService.kt
package com.example.orders

class OrderService(private val repository: OrderRepository) {

    fun place(customerId: String, skus: List<String>): Order {
        val order = repository.create(customerId)
        for (sku in skus) {
            order.addLine(sku)
        }
        return repository.save(order)
    }

    fun find(orderId: String): Order = repository.byId(orderId)
}

File: go/internal/replay/parserfixture/testdata/trees/php/OrderService.php
<?php

namespace Example\Orders;

class OrderService
{
    private OrderRepository $repository;

    public function __construct(OrderRepository $repository)
    {
        $this->repository = $repository;
    }

    public function place(string $customerId, array $skus): Order
    {
        $order = $this->repository->create($customerId);
        foreach ($skus as $sku) {
            $order->addLine($sku);
        }
        return $this->repository->save($order);
    }

    public function find(string $orderId): Order
    {
        return $this->repository->byId($orderId);
    }
}
